# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

### <span style="font-family: Segoe UI, Calibri, sans-serif;">1. Enunciar la hipótesis</span>

<span style="font-family: Segoe UI, Calibri, sans-serif;">
<b>Hipótesis nula (H₀):</b> No hay diferencia en la pérdida de peso promedio entre las dos dietas.<br>
<b>Hipótesis alternativa (H₁):</b> Hay una diferencia en la pérdida de peso promedio entre las dos dietas.
</span>

### 2. Prueba para comprobar la hipótesis

In [1]:
# TODO

import scipy.stats as stats

diet_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

# Prueba t de Student
t_value, p_value = stats.ttest_ind(diet_1, diet_2)

print(f"t-value: {t_value}")
print(f"p-value: {p_value}")

t-value: -3.5383407969933938
p-value: 0.004083270191713912


### 3. Análisis de las conclusiones

- Como el valor p fue menor que el nivel de significancia elegido (5%), se rechaza la hipótesis nula. Esto indica que existe una diferencia significativa en la pérdida de peso entre las dos dietas.

- El valor t negativo indica que el grupo con la segunda dieta presentó una mayor pérdida de peso promedio que el grupo con la primera dieta, lo que sugiere que la segunda dieta podría ser más efectiva para promover la pérdida de peso; sin embargo, este análisis se basa en una muestra limitada, por lo que los resultados deben interpretarse con cautela.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

### 1. Enunciar la hipótesis

<p style="font-family: Arial, sans-serif;">
<b>Hipótesis nula (H₀):</b> Los promedios de rendimiento de maíz de los tres fertilizantes son iguales.<br>
<b>Hipótesis alternativa (H₁):</b> Al menos uno de los promedios de rendimiento de maíz de los fertilizantes es diferente.
</p>

### <span style="font-family: Segoe UI, Calibri, sans-serif;">2. Prueba ANOVA</span>

In [2]:
# TODO

import scipy.stats as stats

fertilizer_1 = [20, 21, 20, 19, 20]
fertilizer_2 = [22, 21, 23, 22, 21]
fertilizer_3 = [24, 23, 22, 23, 24]

# Prueba ANOVA 
f_value, p_value = stats.f_oneway(fertilizer_1, fertilizer_2, fertilizer_3)

print(f"f-value: {f_value}")
print(f"p-value: {p_value}")

f-value: 20.31578947368421
p-value: 0.000140478247931904


### 3. Análisis de las conclusiones

- Dado que el valor p obtenido (0.00014) está muy por debajo del nivel de significancia establecido (0.05), se concluye que hay diferencias significativas en el rendimiento promedio de maíz entre al menos dos de los fertilizantes evaluados, por lo que se rechaza la hipótesis nula.

- El elevado valor f (20.32) indica que la variabilidad entre los grupos (fertilizantes) es considerablemente mayor que la variabilidad dentro de los grupos; esto refuerza la idea de que los tratamientos no son equivalentes y sugiere que ciertos fertilizantes podrían ser más efectivos que otros. Para identificar cuáles, sería necesario aplicar una prueba post hoc como Tukey.

### 4. ¿Cómo podemos saber si un fertilizante es mejor que otro?

Después de realizar una prueba ANOVA y encontrar diferencias significativas entre los grupos, el siguiente paso es realizar una prueba post-hoc, específicamente la prueba de Tukey, para determinar cuál fertilizante es mejor.

Hay dos escenarios posibles al interpretar los resultados de la prueba de Tukey:

1.  Si el primer fertilizante es significativamente diferente del segundo, pero no hay diferencia significativa entre el primero y el tercero, ni entre el segundo y el tercero, podríamos concluir que el primer fertilizante es el mejor o el peor, dependiendo de la dirección de la diferencia.

2.  Si todos los fertilizantes son significativamente diferentes entre sí, entonces se comparan las medias para determinar cuál es el mejor.

In [2]:
import numpy as np
from statsmodels.stats.multicomp import pairwise_tukeyhsd

import scipy.stats as stats

fertilizer_1 = [20, 21, 20, 19, 20]
fertilizer_2 = [22, 21, 23, 22, 21]
fertilizer_3 = [24, 23, 22, 23, 24]

data = np.concatenate([fertilizer_1, fertilizer_2, fertilizer_3])
labels = ["F1"] * 5 + ["F2"] * 5 + ["F3"] * 5

# Prueba de Tukey
result = pairwise_tukeyhsd(data, labels, alpha = 0.05)
print(result)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
    F1     F2      1.8 0.0099 0.4572 3.1428   True
    F1     F3      3.2 0.0001 1.8572 4.5428   True
    F2     F3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


##### ¿Cuál es el mejor fertilizante?

Esta tabla resume los resultados de la prueba de Tukey HSD (Diferencia Honestamente Significativa), que se aplica después de una prueba ANOVA para identificar qué pares de grupos difieren significativamente. Aquí se comparan tres fertilizantes: F1, F2 y F3.

¿Qué significa esto?

- Todas las comparaciones tienen un valor p menor a 0.05, lo que indica que todas las diferencias entre pares de fertilizantes son estadísticamente significativas.  
- El campo `reject = True` confirma que se rechaza la hipótesis nula en cada caso: las medias no son iguales. Si el valor es `True`, significa que hay una diferencia estadísticamente significativa entre esos dos grupos; por lo tanto, debemos mirar la columna `meandiff` para ver cuánto difieren las medias de los dos grupos. Si el valor es positivo, significa que el grupo 2 tiene una media más alta que el grupo 1; si es negativo, es al revés.  
- Los intervalos de confianza no incluyen el cero, lo que refuerza la significancia de las diferencias.

Es decir:

- F3 tiene el mayor rendimiento promedio, ya que:  
- F1 < F3 (diferencia de 3.2)  
- F2 < F3 (diferencia de 1.4)  
- F1 también supera a F2, aunque por una diferencia menor (1.8).

Como todos los fertilizantes son significativamente diferentes entre sí, puedes comparar directamente las medias para determinar cuál es el mejor.

Según los resultados de la prueba de Tukey, todos los fertilizantes presentan diferencias significativas entre sí. El fertilizante F3 muestra el mayor rendimiento promedio, seguido por F1 y luego F2; **por lo tanto, F3 puede considerarse el mejor fertilizante.**